openassessment/data.py

```diff
@@ -1577,3 +1565,125 @@ def get_file_uploads(self, missing_blank=False):
+    scorer_data = user_data_mapping.get(anonymous_scorer_id, {})
+    scorer_name = scorer_data.get('fullname', "Unknown")
+    scorer_username = scorer_data.get('username', "Unknown")
+    scorer_email = scorer_data.get('email', "Unknown")
+    return scorer_name, scorer_username, scorer_email
+
+def generate_assessment_data(assessment_list, user_data_mapping):
+    results = []
+    for assessment in assessment_list:
+
+        scorer_name, scorer_username, scorer_email = get_scorer_data(assessment.scorer_id, user_data_mapping)
+
+        assessment_data = {
+            "id_assessment": str(assessment.id),
+            "scorer_name": scorer_name,
+            "scorer_username": scorer_username,
+            "scorer_email": scorer_email,
+            "assesment_date": assessment.scored_at.strftime('%d-%m-%Y'),
+            "assesment_scores": parts_summary(assessment),
+            "problem_step": score_type_to_string(assessment.score_type),
+            "feedback": assessment.feedback or ''
+        }
+
+        results.append(assessment_data)
+    return results
+
+def generate_received_assessment_data(submission_uuid=None):
+    """
+    Generates a list of received assessments data based on the submission UUID.
+
+    Args:
+        submission_uuid (str, optional): The UUID of the submission. Defaults to None.
+
+    Returns:
+        list[dict]: A list containing assessment data dictionaries.
+    """
+
+    results = []
+
+    submission = None
+    if submission_uuid:
+        submission = sub_api.get_submission_and_student(submission_uuid)
+
+    if not submission:
+        return results
+
+    assessments = _use_read_replica(
+        Assessment.objects.prefetch_related('parts').
+        prefetch_related('rubric').
+        filter(
+            submission_uuid=submission['uuid']
+        )
+    )
+    user_data_mapping = map_anonymized_ids_to_user_data([assessment.scorer_id for assessment in assessments])
+    return generate_assessment_data(assessments, user_data_mapping)
+
+
+def generate_given_assessment_data(item_id=None, submission_uuid=None):
+    """
+    Generates a list of given assessments data based on the submission UUID as scorer.
+
+    Args:
+        submission_uuid (str, optional): The UUID of the submission. Defaults to None.
+
+    Returns:
+        list[dict]: A list containing assessment data dictionaries.
+    """
+    results = []
+    # Getting the scorer student id
+    scorer_submission = sub_api.get_submission_and_student(submission_uuid)
+
+    if not scorer_submission:
+        return results
+
+    scorer_id = scorer_submission['student_item']['student_id']
+    submissions = None
+    if item_id:
+        submissions = Submission.objects.filter(student_item__item_id=item_id).values('uuid')
+        submission_uuids = [sub['uuid'] for sub in submissions]
+
+    if not submission_uuids or not submissions:
+        return results
+
+    # Now fetch all assessments made by this student for these submissions
```

Comment:
which student? we have the context of a scorer but now we're referring to a student